tests/Tests/500__jupyterhub/test-pipChanges-not-permanent.robot

```diff
@@ -0,0 +1,68 @@
+*** Settings ***
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Library          DebugLibrary
+
+Suite Setup      Begin Web Test
+Suite Teardown   End Web Test
+
+
+*** Variables ***
+
+
+*** Test Cases ***
+Open RHODS Dashboard
+  [Tags]  Sanity
+  Wait for RHODS Dashboard to Load
+
+Can Launch Jupyterhub
+  [Tags]  Sanity
+  ${version-check} =  Is RHODS Version Greater Or Equal Than  1.4.0
+  IF  ${version-check}==True
+    Launch JupyterHub From RHODS Dashboard Link
+  ELSE
+    Launch JupyterHub From RHODS Dashboard Dropdown
+  END
+
+Can Login to Jupyterhub
+  [Tags]  Sanity
+  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  ${authorization_required} =  Is Service Account Authorization Required
+  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+
+Can Spawn Notebook
+  [Tags]  Sanity
+  Fix Spawner Status
+  Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook
+
```

Comment:
line 20~39 can be replaced by ### Launch JupyterHub Spawner From Dashboard keyword

Reply:
I tried it but it's not working as expected

Reply:
@milindw96 not sure why it is failing for you i tried on my local machine and it worked. Regarding the other suggestion please see the below dummy code as reference :

_*** Settings ***
Resource         ../../Resources/ODS.robot
Resource         ../../Resources/Common.robot
Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
Suite Setup      Dashboard Test Setup
Suite Teardown   End Web Test

*** Test Cases ***
Can Launch Python3 Notebook And Install Library
  [Tags]  Sanity  ODS-909
  Launch JupyterHub Spawner From Dashboard
  Spawn Notebook With Arguments
  Install Other Python Package     paramiko
  Fix Spawner Status
  Verify Package is not installed    paramiko

*** Keywords ***
Dashboard Test Setup
    [Documentation]    Open browser and load RHODS dashboard
    Set Library Search Order    SeleniumLibrary
    Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
    Login To RHODS Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
    Wait For RHODS Dashboard To Load

Install Other Python Package
   [Documentation]   This keyword intall python packge on Jl notebook using pip
   [Arguments]       ${package_name}
   Add and Run JupyterLab Code Cell in Active Notebook  !pip install ${package_name}
   Add and Run JupyterLab Code Cell in Active Notebook  import ${package_name}
   Capture Page Screenshot
   Wait Until JupyterLab Code Cell Is Not Active
   JupyterLab Code Cell Error Output Should Not Be Visible
   Capture Page Screenshot

Verify Package is not installed
   [Arguments]       ${package_name}
   Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook
   Capture Page Screenshot
   Add and Run JupyterLab Code Cell in Active Notebook  import paramiko
   Wait Until JupyterLab Code Cell Is Not Active
   ${output} =  Get Text  (//div[contains(@class,"jp-OutputArea-output")])[last()]
   Split String    ${output}, " "
   Should Contain  ${output}  ModuleNotFoundError
   Capture Page Screenshot_

Reply:
also if you could share the error via gchat..we can see if it works